web/src/HUD.tsx

```diff
@@ -107,11 +104,17 @@ class HUD extends Component<HudProps, HudState> {
   }
 
   componentDidMount() {
-    this.controller.createNewSocket()
+    if (this.pathBuilder.isSnapshot()) {
+      this.controller.setStateFromSnapshot()
+    } else {
+      this.controller.createNewSocket()
+    }
   }
 
   componentWillUnmount() {
-    this.controller.dispose()
+    if (!this.pathBuilder.isSnapshot()) {
```

Comment:
Maybe there should be a function like `isWebsocket` instead of doing this inverse thing.

Reply:
:+1: